src/detect-engine-port.c

```diff
@@ -779,6 +806,38 @@ int DetectPortJoin(DetectEngineCtx *de_ctx, DetectPort *target,
     return 0;
 }
 
+/**
+ * \brief Checks if two port group lists are equal.
+ *
+ * \param list1 Pointer to the first port group list.
+ * \param list2 Pointer to the second port group list.
+ *
+ * \retval true On success.
+ * \retval false On failure.
+ */
+bool DetectPortListsAreEqual(DetectPort *list1, DetectPort *list2)
+{
+    DetectPort *item = list1;
+    DetectPort *it = list2;
+
+    // First, Check if every item of list1 presents in list2.
+    while (item != NULL && it != NULL) {
+        if (!DetectPortLookupInList(list2, item)) {
```

Comment:
this seems inefficient: instead of comparing each list member, this does a list lookup per member. Why not compare if port at item == it?

Reply:
I assumed that two lists with similar members do not preserve their order in common case, and 'it'  is only used to check if lists sizes differ. Or am I getting something wrong?